core/src/main/java/org/sonarsource/sonarlint/core/util/ProgressReport.java

```diff
@@ -40,7 +41,7 @@ public ProgressReport(String threadName, long period) {
 
   @Override
   public void run() {
-    while (!Thread.interrupted()) {
+    while (!stop) {
```

Comment:
My hypothesis is that the interrupt flag has been somehow lost. So I stopped relying on it for collaborative cancellation. Instead I introduced a `stop` volatile variable.